packages/storage-plus/README.md

```diff
@@ -384,7 +384,172 @@ fn demo() -> StdResult<()> {
+
+    fn key(&self) -> Vec<&[u8]> {
+        let mut keys = self.0.key();
+        keys.extend(&self.1.key());
+        keys
+    }
+}
+```
+
+```rust
+impl<'a> PrimaryKey<'a> for Vec<u8> {
+    type Prefix = ();
+    type SubPrefix = ();
+
+    fn key(&self) -> Vec<&[u8]> {
+        vec![&self]
+    }
+}
+```
+
+---
+
+We can now see how it all works, taking a look at the remaining code:
+
+```rust
+impl<'a> IndexList<TokenInfo> for TokenIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<TokenInfo>> + '_> {
+        let v: Vec<&dyn Index<TokenInfo>> = vec![&self.owner];
+        Box::new(v.into_iter())
+    }
+}
+```
+
+This implements the `IndexList` trait for `TokenIndexes`. This kind of serves two purposes (which are really
+one and the same): it allows the indexes to be queries through `get_indexes`, and, it allows `TokenIndexes`
+to be treated as an `IndexList`. So that it can be passed as a parameter during `IndexedMap` construction, below:
+
+```rust
+pub fn tokens<'a>() -> IndexedMap<'a, &'a str, TokenInfo, TokenIndexes<'a>> {
+    let indexes = TokenIndexes {
+        owner: MultiIndex::new(
+            |d, k| (Vec::from(d.owner.as_ref()), k),
+            "tokens",
+            "tokens__owner",
+        ),
+    };
+    IndexedMap::new("tokens", indexes)
+}
+```
+
+Here `tokens()` is just a helper function, that simplifies the `IndexedMap` construction for us. First the
+index (es) is (are) created, and then, the `IndexedMap` is created (using `IndexedMap::new`), and returned.
+
+During index creation, we must supply an index function per index
+```rust
+        owner: MultiIndex::new(
+            |d, k| (Vec::from(d.owner.as_ref()), k),
+```
+
+, which is the one that will take the value and the primary key (which is always in `Vec<u8>` form) of the
+original map, and create the index key from them. Of course, this requires that the elements required
+for the index key are present in the value (which makes sense).
+Besides the index function, we must also supply the namespace of the pk, and the one for the new index.
+
+---
+
+After that, we just create (and return) the `IndexedMap`:
+
+```rust
+    IndexedMap::new("tokens", indexes)
+```
+
+Here of course, the namespace of the pk must match the one used during index(es) creation. And, we pass our
+`TokenIndexes` (as an `IndexList`-type parameter) as second argument. Connecting in this way the underlying `Map`
+for the pk, with the defined indexes.
+
+So, `IndexedMap` (and the other `Indexed*` types) is just a wrapper / extension around `Map`, that provides
+a number of index functions and namespaces to create indexes over the original `Map` data. It also implements
+calling these index functions during value storage / update / removal, so that you can forget about it,
+and just use the indexed data.
+
+### Usage
+
+An example of use, where `owner` is a `String` value passed as a parameter, and `start_after` and `limit` optionally
```

Comment:
Can you add a simpler example first...

`tokens().idx.owner.get(...)` to show how we can query by owner?
Or is that not possible, as we get a list... and we need to use this?

If so, it would be good to explain the `prefix` before the `range`

Reply:
There's no `get`, as you would have to return a list or iterator.

I can explain `prefix()` / `sub_prefix()` and then show both `range()` and the current `pks()`.